### PR TITLE
Remove unused dependencies from contrib, NIO and NIO examples

### PR DESCRIPTION
**Pull request by jean-philippe-martin**
As per issue #1332:
- move `mockito-core` to test scope
- change `google-cloud-nio` dependency to `google-cloud-storage` rather than `google-cloud`

The google-cloud-nio-example test passes, as expected.

```
$ java -cp google-cloud-contrib/google-cloud-nio/target/google-cloud-nio-0.4.1-SNAPSHOT-shaded.jar:google-cloud-contrib/google-cloud-nio-examples/target/google-cloud-nio-examples-0.4.1-SNAPSHOT.jar com.google.cloud.nio.examples.ListFilesystems
Installed filesystem providers:
  file
  jar
  gs
```

cc: @mziccard

### CODE REVIEW

File: google-cloud-contrib/pom.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
   <dependencies>
     <dependency>
       <groupId>${project.groupId}</groupId>
-      <artifactId>google-cloud</artifactId>
+      <artifactId>google-cloud-storage</artifactId>
```
Comment (mziccard) [inline google-cloud-contrib/pom.xml]: I believe we can get rid of this dependency, no need to import anything here, we do that in child modules. Makes sense?

File: google-cloud-contrib/google-cloud-nio-examples/pom.xml [hunk 2]
```diff
@@ -71,6 +71,7 @@
       <groupId>org.mockito</groupId>
       <artifactId>mockito-core</artifactId>
       <version>1.9.5</version>
+      <scope>test</scope>
```
Comment (mziccard) [inline google-cloud-contrib/google-cloud-nio-examples/pom.xml]: Are test dependencies needed for the example module?

File: google-cloud-contrib/pom.xml (hunk 1 above)
Comment (jean-philippe-martin) [inline google-cloud-contrib/pom.xml]: Indeed! I removed it.

File: google-cloud-contrib/google-cloud-nio-examples/pom.xml (hunk 2 above)
Comment (jean-philippe-martin) [inline google-cloud-contrib/google-cloud-nio-examples/pom.xml]: You're right, I removed all of the test dependencies.

**Comment by mziccard** [comment]
@jean-philippe-martin Thanks for this!

**Comment by jean-philippe-martin** [comment]
Thanks for fixing the title, @mziccard !

**Pull request merged**